### PR TITLE
Filter out older versions when running bwc tests on aarch64

### PR DESCRIPTION
**Pull request by breskeby**
Do not run BWC tests for versions < 7.12 on aarch64 as earlier versions are not supported for this architecture.

### CODE REVIEW

File: buildSrc/src/main/java/org/elasticsearch/gradle/BwcVersions.java [hunk 1]
```diff
@@ -328,15 +330,17 @@ public void compareToAuthoritative(List<Version> authoritativeReleasedVersions)
...
     public List<Version> getIndexCompatible() {
-        return unmodifiableList(
-            Stream.concat(groupByMajor.get(currentVersion.getMajor() - 1).stream(), groupByMajor.get(currentVersion.getMajor()).stream())
-                .collect(Collectors.toList())
-        );
+        var collect = Stream.concat(
+            groupByMajor.get(currentVersion.getMajor() - 1).stream(),
+            groupByMajor.get(currentVersion.getMajor()).stream()
+        ).collect(Collectors.toList());
+        return Architecture.current() == Architecture.AARCH64
```
Comment (mark-vieira) [inline buildSrc/src/main/java/org/elasticsearch/gradle/BwcVersions.java:337]: nit: can we put a newline before the `return`.

File: buildSrc/src/main/java/org/elasticsearch/gradle/BwcVersions.java [hunk 2]
```diff
@@ -345,7 +349,14 @@ public void compareToAuthoritative(List<Version> authoritativeReleasedVersions)
         wireCompat.addAll(groupByMajor.get(currentVersion.getMajor()));
         wireCompat.sort(Version::compareTo);
 
-        return unmodifiableList(wireCompat);
+        return Architecture.current() == Architecture.AARCH64
+            ? unmodifiableList(filterAarch64SupportedVersions(wireCompat))
+            : unmodifiableList(wireCompat);
+    }
+
+    @NotNull
```
Comment (mark-vieira) [inline buildSrc/src/main/java/org/elasticsearch/gradle/BwcVersions.java:357]: Are we doing this consistently? If not we might as well not bother.

**Review by mark-vieira — APPROVED** [review]
LGTM

File: buildSrc/src/main/java/org/elasticsearch/gradle/BwcVersions.java [hunk 3]
```diff
@@ -345,7 +349,14 @@ public void compareToAuthoritative(List<Version> authoritativeReleasedVersions)
...
+    private List<Version> filterAarch64SupportedVersions(List<Version> wireCompat) {
```
Comment (pugnascotia) [inline buildSrc/src/main/java/org/elasticsearch/gradle/BwcVersions.java:358]: What do you think about calling this `filterSupportedVersions` and moving the architecture check here? Then there's only only place where we're checking whether we need to filter.

**Review by pugnascotia — APPROVED** [review]

File: buildSrc/src/main/java/org/elasticsearch/gradle/BwcVersions.java (hunk 1 above)
Comment (breskeby) [reply buildSrc/src/main/java/org/elasticsearch/gradle/BwcVersions.java:336]: done

File: buildSrc/src/main/java/org/elasticsearch/gradle/BwcVersions.java (hunk 2 above)
Comment (breskeby) [reply buildSrc/src/main/java/org/elasticsearch/gradle/BwcVersions.java]: removed. brought to you by idea

File: buildSrc/src/main/java/org/elasticsearch/gradle/BwcVersions.java (hunk 3 above)
Comment (breskeby) [reply buildSrc/src/main/java/org/elasticsearch/gradle/BwcVersions.java:356]: 👍

**Comment by mark-vieira** [comment]
Backported this to 7.12 as well with 2f7f97accf54c9bc16e09c417ff72dd08860bc73.

**Pull request merged**